src/UI/Components/Diagram/actions.ts

```diff
@@ -1,176 +1,104 @@
-import { dia, linkTools } from "@inmanta/rappid";
+import { dia } from "@inmanta/rappid";
 import { DirectedGraph } from "@joint/layout-directed-graph";
+import { uniqueId } from "lodash";
 import { EmbeddedEntity, InstanceAttributeModel, ServiceModel } from "@/Core";
 import { InstanceWithRelations } from "@/Data/Managers/V2/GETTERS/GetInstanceWithRelations";
 import { words } from "@/UI/words";
 import {
-  findCorrespondingId,
-  moveCellFromColliding,
-  toggleLooseElement,
-} from "./helpers";
+  CreateModifierHandler,
+  FieldCreator,
+  createFormState,
+} from "../ServiceInstanceForm";
+import { findCorrespondingId, findFullInterServiceRelations } from "./helpers";
 import activeImage from "./icons/active-icon.svg";
 import candidateImage from "./icons/candidate-icon.svg";
-import { ActionEnum, ConnectionRules, relationId } from "./interfaces";
+import { EmbeddedEventEnum, relationId } from "./interfaces";
 import { Link, ServiceEntityBlock } from "./shapes";
 
+interface Options {
```

Comment:
I presume this will be exported later on to be reusable?  if so, can you make a ticket
In the ticket, it would also be nice to add defaults to the boolean attributes. They can be optional, but the method createComposerEntity should have defaults for all of them.
I think the ticket can be more of a generic action of using the options approach and having the types for the entire Composer helpers methods.

Reply:
Because you export the function that uses this type for its arguments. In all the places where that specific function is used, it could need typing the arguments. Not always, but since it's an object, it can be handy at times :D

Reply:
Oh, right, I thought there is some other factor, I'll just move that to the interfaces with more specific name